### PR TITLE
[csharp] Fix property names

### PR DESCRIPTION
**Pull request by devhl-labs**
This small change fixes multiple issues.

-Underscored variables are usually for private fields. Its pretty unusual to use it for anything else
-Previously, only CodegenModel#vars was getting the property names corrected. That should be fine, but due to the various property collections containing different instances of CodegenProperties (another issue imho) it is not.
-Ensures that the property and the related model constructor parameter have the same name (see the image below). When they are different it can cause deserialization to fail.

image

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.1.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@mandrean (2017/08) @frankyjuang (2019/09) @shibayan (2020/02) @Blackclaws (2021/03) @lucamazzanti (2021/05)

### CODE REVIEW

**Comment by wing328** [comment]
@devhl-labs can you please resolve the merge conflicts when you've time? Then we'll get it merged into 7.0.0 as it'a a breaking change without fallback.

**Comment by devhl-labs** [comment]
Yes, will fix it soon.

**Comment by wing328** [comment]
please resolve the merge conflicts then I'll merge this one first.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractCSharpCodegen.java [hunk 1]
```diff
@@ -492,14 +492,6 @@ public void postProcessModelProperty(CodegenModel model, CodegenProperty propert
...
         for (ModelMap mo : objs.getModels()) {
             CodegenModel cm = mo.getModel();
-            for (CodegenProperty var : cm.vars) {
-                // check to see if model name is same as the property name
-                // which will result in compilation error
-                // if found, prepend with _ to workaround the limitation
-                if (var.name.equalsIgnoreCase(cm.classname)) {
-                    var.name = "_" + var.name;
-                }
-            }
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractCSharpCodegen.java:502]: I think we may need to keep this code block and replace "_" with "var_" instead. We'll see after merging this PR.

Comment (wing328) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractCSharpCodegen.java:502]: Appveyor CI tests passed. Confirmed there's no issue.

**Pull request merged**